nautobot/core/models/name_color_content_types.py

```diff
@@ -0,0 +1,88 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.urls import reverse
+
+from nautobot.core.fields import AutoSlugField
+from nautobot.core.models import BaseModel
+from nautobot.extras.models.customfields import CustomFieldModel
+from nautobot.extras.models.change_logging import ChangeLoggedModel
+from nautobot.extras.models.relationships import RelationshipModel
+from nautobot.extras.models.mixins import DynamicGroupMixin, NotesMixin
+from nautobot.utilities.choices import ColorChoices
+from nautobot.utilities.fields import ColorField
+from nautobot.utilities.querysets import RestrictedQuerySet
+
+
+class ContentTypeRelatedQuerySet(RestrictedQuerySet):
+    def get_for_model(self, model):
+        """
+        Return all `self.model` instances assigned to the given model.
+        """
+        content_type = ContentType.objects.get_for_model(model._meta.concrete_model)
+        return self.filter(content_types=content_type)
+
+    def get_by_natural_key(self, name):
+        return self.get(name=name)
+
+
+# Inheriting from OrganizationalModel here causes partial import error
```

Comment:
Can we put a TODO here?

Alternately, could we make this a mixin instead of a base class, and thus have Role defined as `class Role(NameColorContentTypesMixin, OrganizationalModel):`?

Reply:
Importing `OrganizationalModel` in status / role causes partial import error